app/gen-server/lib/HomeDBManager.ts

```diff
@@ -4745,6 +4727,35 @@ export class HomeDBManager extends EventEmitter {
     if (thisUser) { users.push(thisUser); }
     return { personal: true, public: !realAccess };
   }
+
+  private _getWorkspaceWithACLRules(scope: Scope, wsId: number) {
+    const query = this._workspace(scope, wsId, {
+      markPermissions: Permissions.VIEW
+    })
+    // Join the workspace's ACL rules (with 1st level groups/users listed).
+    .leftJoinAndSelect('workspaces.aclRules', 'acl_rules')
+    .leftJoinAndSelect('acl_rules.group', 'workspace_groups')
+    .leftJoinAndSelect('workspace_groups.memberUsers', 'workspace_group_users')
+    .leftJoinAndSelect('workspace_groups.memberGroups', 'workspace_group_groups')
+    .leftJoinAndSelect('workspace_group_users.logins', 'workspace_user_logins')
+    .leftJoinAndSelect('workspaces.org', 'org');
+    return verifyIsPermitted(query);
+  }
+
+  private _getOrgWithACLRules(scope: Scope, org: number|string) {
+    const orgQuery = this.org(scope, org, {
+      markPermissions: Permissions.VIEW,
+      needRealOrg: true,
+      allowSpecialPermit: true
```

Comment:
I still have doubts on this part: we not only control the permissions for the groupes associated with the workspaces, but also now the permissions for the orgs.

Otherwise, I may remove the call to `verifyIsPermitted` and these options when we call that method for the workspaces, if that makes sense.

Reply:
That sounds right. Makes sense to me to exclude them when called from that method.

Reply:
I ended up renaming `verifyIsPermitted` to `verifyEntity` to allow discarding the permission check.

https://github.com/gristlabs/grist-core/pull/824/commits/f8bed0b689522058aabb6506efcafc46d192e8c3

I felt like we may have wanted dealing with monad in that function (like [`Result`](https://doc.rust-lang.org/std/result/) in Rust), or throw exceptions that would be handled in the express route that would deduce the status code to return.